src/python/WMCore/GlobalWorkQueue/CherryPyThreads/HeartbeatMonitor.py

```diff
@@ -1,34 +1,180 @@
+            doc = dict()
+            doc["type"] = "work_prio_status"
+            doc["status"] = item['status']
+            doc["priority"] = item['priority']
+            doc["count"] = item['count']  # total number of workqueue elements
+            doc["sum"] = item['sum']  # total number of top level jobs
+            doc.update(commonInfo)
+            docs.append(doc)
+
+        self._fillMissingStatus(stats['workByAgentAndStatus'])
+        for item in stats['workByAgentAndStatus']:
+            doc = dict()
+            doc["type"] = "work_agent_status"
+            doc["status"] = item['status']
+            doc["agent_name"] = item['agent_name']
+            doc["count"] = item['count']  # total number of workqueue elements
+            doc["sum"] = item['sum']  # total number of top level jobs
+            doc.update(commonInfo)
+            docs.append(doc)
+
+        for item in stats['workByAgentAndPriority']:
+            doc = dict()
+            doc["type"] = "work_agent_prio"
+            doc["priority"] = item['priority']
+            doc["agent_name"] = item['agent_name']
+            doc["count"] = item['count']  # total number of workqueue elements
+            doc["sum"] = item['sum']  # total number of top level jobs
+            doc.update(commonInfo)
+            docs.append(doc)
+
+        # jobs respecting data location constraints
+        for status, items in stats['uniqueJobsPerSite'].iteritems():
+            for item in items:
+                doc = {}
+                doc["type"] = "work_site_unique"
+                doc["status"] = status
+                doc.update(commonInfo)
+                doc.update(item)
+                docs.append(doc)
+        for status, items in stats['possibleJobsPerSite'].iteritems():
+            for item in items:
+                doc = {}
+                doc["type"] = "work_site_possible"
+                doc["status"] = status
+                doc.update(commonInfo)
+                doc.update(item)
+                docs.append(doc)
+
+        # jobs NOT respecting any data location, so assuming they can run anywhere on the sitewhitelist
+        for status, items in stats['uniqueJobsPerSiteAAA'].iteritems():
+            for item in items:
+                doc = {}
+                doc["type"] = "work_site_uniqueAAA"
+                doc["status"] = status
+                doc.update(commonInfo)
+                doc.update(item)
+                docs.append(doc)
+        for status, items in stats['possibleJobsPerSiteAAA'].iteritems():
+            for item in items:
+                doc = {}
+                doc["type"] = "work_site_possibleAAA"
+                doc["status"] = status
+                doc.update(commonInfo)
+                doc.update(item)
+                docs.append(doc)
+
+        self.logger.info("%i docs created to post to MonIT", len(docs))
+        return docs
+
+    def uploadToAMQ(self, docs, producer="global_workqueue"):
```

Comment:
This seems to be exactly the same method as uploadToAMQ in reqmgr2 heartbeat monitor except producer. Maybe we should move this is base class?